eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -1430,6 +1430,94 @@ public void value(K key, V value)
         return builder.toString();
     }
 
+    public boolean trimToSize()
+    {
+        if (this.table.length <= this.occupied)
```

Comment:
So, if I am not mistaken, the short-circuit should be:

`if(this.table.length/2 <= this.occupied)`

Similarly: 
`if (this.table.length/2 <= defaultSizeToAllocate)`

and then:

`this.init(Math.min(defaultSizeToAllocate, this.table.length))`